programs/account-compression/src/instructions/nullify_leaves.rs

```diff
@@ -99,8 +99,8 @@ fn insert_nullifier(
     let seq = (merkle_tree.sequence_number + 1) as u64;
     for (i, leaf_queue_index) in leaves_queue_indices.iter().enumerate() {
         let leaf_cell = nullifier_queue
-            .by_value_index(*leaf_queue_index as usize, None)
-            .cloned()
+            .get_bucket(*leaf_queue_index as usize)
+            .ok_or(AccountCompressionErrorCode::LeafNotFound)?
```

Comment:
We need a check here that the sequence number of that hash cell is None.
So far we relied in the check that by_value_index returned only hash cells with None as sequence number but get_bucked doesn't have this check anymore.
I see three options:
1. add the option parameter to get_bucket
2. create a method get_valid_bucket(index, Option<sequence_number>)
3. add a manual check that sequence number is None in insert_nullifier, update_address_merkle_tree and test_forester (possibly other places)

Reply:
I like the option 2.